### PR TITLE
feat: platform support for oras pull and oras copy

### LINKED ISSUE #530: Platform support for `oras pull` and `oras copy`
Should we support `--platform` for `oras pull` and `oras copy`?

Actual:
```
$ oras pull ghcr.io/homebrew/core/hello:2.12.1
Downloading 62534bceb8f7 hello--2.12.1.monterey.bottle.tar.gz
Downloading a0103553329c hello--2.12.1.arm64_monterey.bottle.tar.gz
Downloading e8c7459b0310 hello--2.12.1.arm64_big_sur.bottle.tar.gz
Downloaded  a0103553329c hello--2.12.1.arm64_monterey.bottle.tar.gz
Downloading 480a77f0f4e0 hello--2.12.1.big_sur.bottle.tar.gz
Downloaded  62534bceb8f7 hello--2.12.1.monterey.bottle.tar.gz
Downloading c30c2be3191b hello--2.12.1.catalina.bottle.tar.gz
Downloaded  e8c7459b0310 hello--2.12.1.arm64_big_sur.bottle.tar.gz
Downloading 7935d0efdae6 hello--2.12.1.x86_64_linux.bottle.tar.gz
Downloaded  480a77f0f4e0 hello--2.12.1.big_sur.bottle.tar.gz
Downloaded  c30c2be3191b hello--2.12.1.catalina.bottle.tar.gz
Downloaded  7935d0efdae6 hello--2.12.1.x86_64_linux.bottle.tar.gz
Pulled ghcr.io/homebrew/core/hello:2.12.1
Digest: sha256:f902e04beb3576e8db2ad3606324ecd675298f672f8f73eb983800342d757092
```

Expected:
```
$ oras pull ghcr.io/homebrew/core/hello:2.12.1 --platform "darwin/amd64:macOS 11.6"
Downloading c30c2be3191b hello--2.12.1.catalina.bottle.tar.gz
Downloaded  c30c2be3191b hello--2.12.1.catalina.bottle.tar.gz
Pulled ghcr.io/homebrew/core/hello:2.12.1
Digest: sha256:c026d71d90e009e8d19b761d5efeb3258981d0adff50c24a04c35020a2e947d0
```

### PR DESCRIPTION
**Pull request by jasminetMSFT**
Resolves #530 
Signed-off-by: JasmineTang <jasminetang@microsoft.com>

### CODE REVIEW

File: cmd/oras/copy.go [hunk 1]
```diff
@@ -124,6 +132,7 @@ func runCopy(opts copyOptions) error {
 			copyOptions := oras.CopyOptions{
 				CopyGraphOptions: extendedCopyOptions.CopyGraphOptions,
 			}
+			copyOptions.WithTargetPlatform(targetPlatform)
```
Comment (lizMSFT) [inline cmd/oras/copy.go:135]:
```suggestion
			if targetPlatform != nil {
				copyOptions.WithTargetPlatform(targetPlatform)
			}
```

Comment (qweeah) [reply cmd/oras/copy.go:135]: What does `nil` represent?

Comment (Wwwsylvia) [reply cmd/oras/copy.go:135]: Yeah `nil` should be considered as a `no input`. We should fix this in `oras-go`.

Comment (Wwwsylvia) [reply cmd/oras/copy.go:135]: Issue created: https://github.com/oras-project/oras-go/issues/318

File: cmd/oras/copy.go [hunk 2]
```diff
@@ -53,6 +54,9 @@ Example - Copy the artifact tagged 'v1' from repository 'localhost:5000/net-moni
 
 Example - Copy the artifact tagged 'v1' and its referrers from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy'
   oras cp -r localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1
+
+Example - Copy the artifact tagged 'v1' from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy' with certain platform
+  oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1 --platform linux/arm/v5
```
Comment (shizhMSFT) [inline cmd/oras/copy.go:59]: Please move the argument before parameters.

File: cmd/oras/pull.go [hunk 3]
```diff
@@ -65,6 +66,9 @@ Example - Pull files from the HTTP registry:
 Example - Pull files with local cache:
   export ORAS_CACHE=~/.oras/cache
   oras pull localhost:5000/hello:latest
+
+Example - Pull files with certain platform:
+  oras pull localhost:5000/hello:latest --platform linux/arm/v5
```
Comment (shizhMSFT) [inline cmd/oras/pull.go:71]: same here

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**